callouts/java/service-callout/src/main/java/service/ServiceCalloutTools.java

```diff
@@ -0,0 +1,166 @@
+        if (add != null) {
+            headerBuilder.addAllSetHeaders(add);
+        }
+        if (remove != null) {
+            headerBuilder.addAllRemoveHeaders(remove);
+        }
+        if (clearRouteCache != null) {
+            responseBuilder.setClearRouteCache(clearRouteCache);
+        }
+        return headersResponseBuilder.build();
+    }
+
+    /**
+     * Builds a body mutation response.
+     *
+     * @param bodyResponseBuilder Builder for modifying response body
+     * @param body                The body content to be set
+     * @param clearBody           Boolean indicating whether to clear the body
+     * @param clearRouteCache     Boolean indicating whether to clear the route cache
+     * @return Constructed BodyResponse object
+     */
+    public static BodyResponse AddBodyMutations(
+            BodyResponse.Builder bodyResponseBuilder,
+            String body,
+            Boolean clearBody,
+            Boolean clearRouteCache) {
+        CommonResponse.Builder responseBuilder = bodyResponseBuilder.getResponseBuilder();
+        BodyMutation.Builder bodyBuilder = responseBuilder.getBodyMutationBuilder();
+        if (body != null) {
+            bodyBuilder.setBody(ByteString.copyFromUtf8(body));
+        }
+        if (clearBody != null) {
+            bodyBuilder.setClearBody(clearBody);
+        }
+        if (clearRouteCache != null) {
+            responseBuilder.setClearRouteCache(clearRouteCache);
+        }
+        return bodyResponseBuilder.build();
+    }
+
+
+    /**
+     * Creates an immediate HTTP response with specific headers and status code.
+     * <p>
+     * Args:
+     * code (StatusCode): The HTTP status code to return.
+     * headers: Optional list of tuples (header, value) to include in the response.
+     * append_action: Optional action specifying how headers should be appended.
+     * <p>
+     * Returns:
+     * ImmediateResponse: Configured immediate response with the specified headers and status code.
+     */
+//    public static ImmediateResponse BuildImmediateResponse (
+//            ImmediateResponse.Builder immediateResponseBuilder,
+//            Iterable<Map.Entry<String, String>> addHeaders,
+//            HttpStatus status
+//    ) {
+//
+//        HeaderMutation.Builder headerMutationBuilder = immediateResponseBuilder
+//                .getHeadersBuilder();
+//
+//        for(Map.Entry<String, String> entry : addHeaders) {
+//            headerMutationBuilder
+//                    .addSetHeadersBuilder()
+//                    .getHeaderBuilder()
+//                    .setKey(entry.getKey())
+//                    .setValue(entry.getValue())
+//                    .setRawValue(ByteString.copyFromUtf8(entry.getValue()));
+//        }
+//
+//        immediateResponseBuilder.setStatus(status);
+//
+//        return immediateResponseBuilder.build();
+//
+//    }
+    public static ImmediateResponse BuildImmediateResponse(
```

Comment:
This should just return the builder like before. It should also take in a builder rather than creating one.

Reply:
This method should really be replaced with the above one.